File: user-service/src/main/java/rs/edu/raf/service/EmployeeService.java
package rs.edu.raf.service;

import rs.edu.raf.dto.EditEmployeeDTO;
import rs.edu.raf.dto.EmployeeCreateDTO;
import rs.edu.raf.dto.EmployeeDTO;

import java.util.List;

public interface EmployeeService {
    /**
     * Creates a new employee.
     *
     * @param employeeCreateDTO The DTO containing information about the new employee.
     * @return The created {@link EmployeeDTO} object.
     */
    EmployeeDTO createNewEmployee(EmployeeCreateDTO employeeCreateDTO);

    /**
     * Edits an employee.
     *
     * @param editEmployeeDTO The DTO containing information to edit the employee.
     * @return The updated {@link EmployeeDTO} object.
     */
    EmployeeDTO editEmployee(EditEmployeeDTO editEmployeeDTO);

    /**
     * Lists all active employees.
     *
     * @return A list of {@link EmployeeDTO} objects representing all active employees.
     */
    List<EmployeeDTO> getAllActiveEmployees();

    /**
     * Finds an active employee by email.
     *
     * @param email The email address of the worker to find.
     * @return The {@link EmployeeDTO} object representing the found employee.
     */
    EmployeeDTO findActiveEmployeeByEmail(String email);

    /**
     * Finds an active employee by id
     * @param id the id of the employee
     * @return The {@link EmployeeDTO} object representing the found employee
     */
    EmployeeDTO findActiveEmployeeById(Long id);
}

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/service/EmployeeService.kt
package org.example.userservicekotlin.service

import org.example.userservicekotlin.dto.EditEmployeeDTO
import org.example.userservicekotlin.dto.EmployeeCreateDTO
import org.example.userservicekotlin.dto.EmployeeDTO

interface EmployeeService {
    /**
     * Creates a new employee.
     *
     * @param employeeCreateDTO The DTO containing information about the new employee.
     * @return The created [EmployeeDTO] object.
     */
    fun createNewEmployee(employeeCreateDTO: EmployeeCreateDTO): EmployeeDTO

    /**
     * Edits an employee.
     *
     * @param editEmployeeDTO The DTO containing information to edit the employee.
     * @return The updated [EmployeeDTO] object.
     */
    fun editEmployee(editEmployeeDTO: EditEmployeeDTO): EmployeeDTO

    /**
     * Lists all active employees.
     *
     * @return A list of [EmployeeDTO] objects representing all active employees.
     */
    fun getAllActiveEmployees(): List<EmployeeDTO>

    /**
     * Finds an active employee by email.
     *
     * @param email The email address of the worker to find.
     * @return The [EmployeeDTO] object representing the found employee.
     */
    fun findActiveEmployeeByEmail(email: String): EmployeeDTO

    /**
     * Finds an active employee by id
     * @param id the id of the employee
     * @return The [EmployeeDTO] object representing the found employee
     */
    fun findActiveEmployeeById(id: Long): EmployeeDTO
}